chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/product/MergeActionPermissions.java

```diff
@@ -1,20 +1,42 @@
 package gov.healthit.chpl.permissions.domains.product;
 
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import gov.healthit.chpl.dao.ProductDAO;
+import gov.healthit.chpl.dto.ProductDTO;
 import gov.healthit.chpl.permissions.domains.ActionPermissions;
 
 @Component("productMergeActionPermissions")
 public class MergeActionPermissions extends ActionPermissions {
 
+    @Autowired
+    private ProductDAO productDAO;
+
     @Override
     public boolean hasAccess() {
-        return getResourcePermissions().isUserRoleAdmin() || getResourcePermissions().isUserRoleOnc();
+        return false;
     }
 
     @Override
     public boolean hasAccess(Object obj) {
-        return false;
+        if (getResourcePermissions().isUserRoleAdmin() || getResourcePermissions().isUserRoleOnc()) {
+            return true;
+        } else if (getResourcePermissions().isUserRoleAcbAdmin()) {
+            try {
+                // Need to make sure developer for products to be merged is active
+                List<Long> productIds = (List<Long>) obj;
+                // All products will have the same developer, so we only need to check the first one
```

Comment:
I guess I'm just wondering if this comment is true... I think probably the way the UI lets you do a product merge it is true but if I just call the API with some random product ids what would happen?

Reply:
Good catch Katy. Definitely see that as a possible issue

Reply:
I'll add to check each developer for a product.

Reply:
Sounds good.. make sure to do the same thing for the versions too as I think it had a similar comment.